### PR TITLE
mariadb: use fakeroot instead of postinst script

### PR DESCRIPTION
**Pull request by micmac1**
fakeroot support was recently added to OpenWrt. Let's make use of it.

Signed-off-by: Sebastian Kemper <sebastian_ml@gmx.net>

Maintainer: @miska @BKPepe 
Compile tested: master ath79
Run tested: 19.07 ath79, installed package, checked that permissions are as expected.

Description:
This removes postinst script in favor of using fakeroot.

Kind regards,
Seb

### CODE REVIEW

**Comment by aparcar** [comment]
You can now use `mariadb` instead of a numeric value. Please update this PR

**Comment by aparcar** [comment]
Great thanks, I'm not the maintainer here so won't merge it. But I approve the change :)

**Review by aparcar — APPROVED** [review]

**Review by BKPepe — APPROVED** [review]

**Comment by miska** [comment]
Looks great, thanks!

**Review by miska — APPROVED** [review]

**Comment by neheb** [comment]
https://downloads.openwrt.org/snapshots/faillogs/arm_mpcore_vfp/packages/mariadb/compile.txt

Related?

**Comment by aparcar** [comment]
Could it be a possible problem that the variables `type` and `name` are declared as `local` and then called in a subshell?

https://github.com/openwrt/openwrt/blob/master/scripts/ipkg-build#L79

**Comment by micmac1** [comment]
> Could it be a possible problem that the variables `type` and `name` are declared as `local` and then called in a subshell?
>
> https://github.com/openwrt/openwrt/blob/master/scripts/ipkg-build#L79

Hi guys,

I checked all targets today and only found arm_mpcore_vfp to show this
problem. I *think* the SDK used for this target is
openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64. So I
downloaded it and tried to reproduce. But there was no failure there.

It came to this point:

/home/sk/tmp/sdk/openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64/staging_dir/host/bin/fakeroot -l /home/sk/tmp/sdk/openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64/staging_dir/host/lib/libfakeroot.so -f /home/sk/tmp/sdk/openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64/staging_dir/host/bin/faked /home/sk/tmp/sdk/openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64/scripts/ipkg-build -m "/usr/lib/mariadb/plugin/auth_pam_tool_dir:root:mariadb:0750" /home/sk/tmp/sdk/openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64/build_dir/target-arm_mpcore_musl_eabi/mariadb-10.4.14/ipkg-arm_mpcore/mariadb-server-plugin-auth-pam /home/sk/tmp/sdk/openwrt-sdk-oxnas-ox820_gcc-8.4.0_musl_eabi.Linux-x86_64/bin/packages/arm_mpcore/packages

And then it happily continued. So maybe the issue is how this SDK is set
up on the build bots? I think this may only be debugged by somebody with
access to the build bots.

So I think this cannot have anything to do with subshells etc., because
then I'd expect it to fail for other targets likewise, which is not the
case.

Regards,
Seb

**Pull request merged**
